recipes/libmodbus/all/conanfile.py

```diff
@@ -23,101 +29,116 @@ class LibmodbusConan(ConanFile):
         "fPIC": True,
     }
 
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _settings_build(self):
-        return self.settings_build if hasattr(self, "settings_build") else self.settings
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            try:
+                del self.options.fPIC # once removed by config_options, need try..except for a second del
+            except Exception:
+                pass
+        try:
+            del self.settings.compiler.libcxx # for plain C projects only
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd # for plain C projects only
+        except Exception:
+            pass
 
     def build_requirements(self):
-        if self.settings.compiler == "Visual Studio":
-            self.build_requires("automake/1.16.3")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        if self.settings.os == "Macos" or self.settings.compiler == "Visual Studio":
```

Comment:
```suggestion
        if self.settings.os == "Macos" or is_msvc(self):
```
from conan.tools.microsoft import is_msvc. It checks both Visual Studio and msvc compiler.